kview/src/main/java/dev/ikm/komet/kview/mvvm/view/pattern/PatternDetailsController.java

```diff
@@ -746,38 +839,224 @@ private void addPaneToTray(Pane contentViewPane, Pane slideoutTrayPane) {
+
+                    PatternEntity<PatternEntityVersion> patternEntity = semanticEntity.pattern();
+                    PatternEntityVersion patternEntityVersion = viewCalculator.latest(patternEntity).get();
+
+                    int indexForDescrType = patternEntityVersion.indexForMeaning(TinkarTerm.DESCRIPTION_TYPE);
+
+                    // Filter (include) semantics where they contain descr type having FQN, Regular name, Definition Descr.
+                    Object descriptionTypeConceptValue = semanticVersion.get().fieldValues().get(indexForDescrType);
+                    if(descriptionTypeConceptValue instanceof EntityFacade descriptionTypeConcept ){
+                        int typeId = descriptionTypeConcept.nid();
+                        return (typeId == FULLY_QUALIFIED_NAME_DESCRIPTION_TYPE.nid() ||
+                                typeId == REGULAR_NAME_DESCRIPTION_TYPE.nid() ||
+                                typeId == DEFINITION_DESCRIPTION_TYPE.nid());
+                    }
+                    return false;
+                }).forEach(semanticEntity -> {
+                    // Each description obtain the latest semantic version, pattern version and their field values based on index
+                    Latest<SemanticEntityVersion> semanticVersion = viewCalculator.latest(semanticEntity);
+                    PatternEntity<PatternEntityVersion> patternEntity = semanticEntity.pattern();
+                    PatternEntityVersion patternEntityVersion = viewCalculator.latest(patternEntity).get();
+
+                    int indexCaseSig = patternEntityVersion.indexForMeaning(DESCRIPTION_CASE_SIGNIFICANCE);
+                    int indexLang = patternEntityVersion.indexForMeaning(LANGUAGE_CONCEPT_NID_FOR_DESCRIPTION);
+
+                    List<String> descrFields = new ArrayList<>();
+                    descriptionSemanticsMap.put(semanticVersion.get(), descrFields);
+                    Object caseSigConcept = semanticVersion.get().fieldValues().get(indexCaseSig);
+                    Object langConcept = semanticVersion.get().fieldValues().get(indexLang);
+
+                    // e.g. FQN - English | Case Sensitive
+                    String casSigText = viewCalculator.getRegularDescriptionText(((ConceptFacade) caseSigConcept).nid())
+                            .orElse(String.valueOf(((ConceptFacade) caseSigConcept).nid()));
+                    String langText = viewCalculator.getRegularDescriptionText(((ConceptFacade) langConcept).nid())
+                            .orElse(String.valueOf(((ConceptFacade) langConcept).nid()));
+
+                    descrFields.add(casSigText);
+                    descrFields.add(langText);
+                });
+        return descriptionSemanticsMap;
+
+    }
+
+
     @FXML
     private void savePattern(ActionEvent actionEvent) {
         boolean isValidSave = patternViewModel.createPattern();
         LOG.info(isValidSave ? "success" : "failed");
```

Comment:
Note: see cognitives new features to know if something is invalid.
https://github.com/carldea/cognitive/wiki#the-invalid-property-of-a-validationviewmodel-validationviewmodels-invalidproperty-method

Reply:
the comment you are making is code that is not part of the PR... I recommend we move to a refactoring ticket